core/templates/services/math-interactions.service.ts

```diff
@@ -107,16 +109,30 @@ export class MathInteractionsService {
       return false;
     }
     try {
+      expressionString = this.insertMultiplicationSigns(expressionString);
       expressionObject = nerdamer(expressionString);
     } catch (err) {
       this.warningText = this.cleanErrorMessage(err.message, expressionString);
       return false;
     }
-    if (algebraic && expressionObject.variables().length === 0) {
-      this.warningText = 'It looks like you have entered only ' +
+    if (algebraic) {
```

Comment:
Extract just the common parts into _validateExpression() and name that function accordingly. Algebraic/numeric-specific stuff should be dealt with in the individual functions.

Reply:
Done.